asdf/_config.py

```diff
@@ -0,0 +1,185 @@
+"""
```

Comment:
This module changed a bit from the version that was merged into the `asdf-standard-2.0.0` branch.  I ended up wanting to make the config mutable after all, so that developers could add resources at runtime from the console.  This ended up simplifying the method signatures a good bit.